sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -60,22 +60,10 @@ object SimpleAnalyzer extends Analyzer(
   },
   new SQLConf().copy(SQLConf.CASE_SENSITIVE -> true))
 
-object FakeV2SessionCatalog extends TableCatalog {
+object FakeV2SessionCatalog extends CatalogPlugin {
   private def fail() = throw new UnsupportedOperationException
-  override def listTables(namespace: Array[String]): Array[Identifier] = fail()
-  override def loadTable(ident: Identifier): Table = {
-    throw new NoSuchTableException(ident.toString)
-  }
-  override def createTable(
-      ident: Identifier,
-      schema: StructType,
-      partitions: Array[Transform],
-      properties: util.Map[String, String]): Table = fail()
-  override def alterTable(ident: Identifier, changes: TableChange*): Table = fail()
-  override def dropTable(ident: Identifier): Boolean = fail()
-  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = fail()
   override def initialize(name: String, options: CaseInsensitiveStringMap): Unit = fail()
-  override def name(): String = fail()
+  override def name(): String = "fake_v2_session"
```

Comment:
I think the name should be `CatalogManager.SESSION_CATALOG_NAME`

Reply:
This can be anything, no? I can change it, but curious why it has to be `CatalogManager.SESSION_CATALOG_NAME`

Reply:
so that you don't need to rely on the fake name at https://github.com/apache/spark/pull/25771/files#diff-c3084f55edbba87c6eca641c135a2e79R32

Reply:
OK, I will change it. I thought it was more clear this way to indicate that `FakeV2SessionCatalog` was being used instead of `V2SessionCatalog`.